pkg/controller/config/dns.go

```diff
@@ -97,50 +147,83 @@ func (h *configHandler) Handle(req router.Request, resp router.Response) error {
 		sec.Data = map[string][]byte{"domain": []byte(domain), "token": []byte(token)}
 		if err := req.Client.Update(req.Ctx, sec); err != nil {
-			return err
+			return "", err
 		}
 	}
 
 	if !strings.EqualFold(*cfg.LetsEncrypt, "disabled") {
 		if err := tls.ProvisionWildcardCert(req, resp, domain, token); err != nil {
-			return err
+			return "", err
 		}
 	}
 
-	return nil
+	return domain, nil
 }
 
-// purgeRecordsIfDisabling checks if we are transitioning AcorDNS from an enabled state to a disabled state and if so calls the
-// acorn DNS service to purge all records for the domain. It is expected that string variable returned by this function
-// will be set as the labels.AcornDNSState annotation on the acorn-dns secret
-func purgeRecordsIfDisabling(req router.Request, domain string, cfg *apiv1.Config, dnsSecret *corev1.Secret,
-	token string, dnsClient dns.Client) (string, error) {
-	var state string
-	if domain == "" {
-		state = *cfg.AcornDNS
-	} else {
-		// The config object includes the acorn domain in the list of ClusterDomains if we are in an enabled state.
-		if slices.Contains(cfg.ClusterDomains, domain) {
-			state = "enabled"
-		} else {
-			state = "disabled"
-		}
+func (h *configHandler) handleIngress(req router.Request, resp router.Response, domain string, acornDNS *string, ingressClassName *string) error {
+	if acornDNS == nil || *acornDNS == "disabled" {
+		return nil
 	}
 
-	// purge the records if the current state is disabled, but that isn't the state recorded on the dnsSecret
-	// If this is the first time the handler is being called, then the domain and token will be blank and thus we
-	// shouldn't call purge
-	if strings.EqualFold(state, "disabled") && dnsSecret.Annotations[labels.AcornDNSState] != "disabled" {
-		if domain != "" && token != "" {
-			if err := dnsClient.PurgeRecords(*cfg.AcornDNSEndpoint, domain, token); err != nil {
-				if dns.IsDomainAuthError(err) {
-					if err := dns.ClearDNSToken(req.Ctx, req.Client, dnsSecret); err != nil {
-						return "", err
-					}
-				}
-				return "", err
-			}
+	var err error
+	if ingressClassName == nil {
+		ingressClassName, err = publish.IngressClassNameIfNoDefault(req.Ctx, req.Client)
+		if err != nil {
+			return err
 		}
 	}
-	return state, nil
+
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      system.ServiceName,
+			Namespace: system.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Port: 80,
+			}},
+			Selector: map[string]string{
+				"app": system.Namespace,
+			},
+		},
+	}
+
+	pt := netv1.PathTypeImplementationSpecific
+	ingress := &netv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: system.IngressName,
+			Labels: map[string]string{
+				labels.AcornManaged: "true",
+			},
+		},
+		Spec: netv1.IngressSpec{
+			IngressClassName: ingressClassName,
+			Rules: []netv1.IngressRule{
+				{
+					Host: strings.TrimPrefix(domain, "."), // Needed since domain starts with a "."
```

Comment:
this sets the host to something like `x96fr6.oss-acorn.io`. Can we do a sub domain of that? like `dns-test.x96fr6.oss-acorn.io`. That would let us potentially dig that that domain and prove its resolving properly.

Reply:
Oh I see, you did this because later in ToRecordRequestsAndHash, if the host matches the clusterDomain exactly, you make the record be `*`.

...maybe we should change the impl of ToRecordRequestsAndHash...more on that down there in that function

Reply:
Ah, yeah I like that better. Good call.